codice/ddf
# DDF-4650 Add matchCase support to WFS 1.1 filter delegate

#### What does this PR do?
Updates the WFS 1.1.0 filter delegate to set the `matchCase` attribute appropriately when the query should be case-sensitive or case-insensitive.

#### Who is reviewing it? 
@brianfelix 
@Corey-Collins 
@leo-sakh 
@samuelechu 
@kentmorrissey 

#### Select relevant component teams: 
@codice/io 
@codice/ogc 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@troymohl

#### How should this be tested?
1. Create a WFS 1.1.0 source with the URL http://services.azgs.az.gov/ArcGIS/services/aasggeothermal/AZWellLogs/MapServer/WFSServer
a. Disable the CN check and allow redirects. Leave everything else as-is.
2. In the Karaf console, run `log:set info org.apache.cxf` and `log:set warn org.apache.camel`.
3. Open Intrigue.
4. Test different kinds of textual queries with the advanced query builder against the WFS source and verify the `matchCase` attribute is correct for each:
a. To check the correctness of a query, open the mock server console, find the most recent `GetFeature` request, and verify its filter XML.
b. `CONTAINS` should map to a `PropertyIsLike` with `matchCase="false"`
c. `MATCHCASE` should map to a `PropertyIsLike` with `matchCase="true"`
d. `=` should map to a `PropertyIsEqualTo` with `matchCase="true"`
e. Make sure to test both `anyText` queries and queries specifying individual attributes. You can query on any of the fields from http://services.azgs.az.gov/ArcGIS/services/aasggeothermal/AZWellLogs/MapServer/WFSServer?version=1.1.0&service=WFS&request=DescribeFeatureType with a type of `xsd:string`. The corresponding metacard attribute you would query on is `ext.AZWellLogs.<element name>`, e.g. `ext.AZWellLogs.DisplayName`.
f. Make sure to test different types of textual queries combined with `AND`, `OR`, `NOT AND` and `NOT OR`, e.g. `ext.AZWellLogs.Source CONTAINS 'anything' AND ext.AZWellLogs.Notes = 'something'`

#### Any background context you want to provide?
#### What are the relevant tickets?
Fixes: #4650 

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
